I have contributed this fix because of a syntax error, that i has to be declared.
I have tested it in our oppenSUSE build environment.

The new failure for PowerPC and s390x is the following:
```
[ 2131s] /usr/lib64/gcc/s390x-suse-linux/11/../../../../s390x-suse-linux/bin/ld: opencl_7z_fmt_plug.o: in function `sevenzip_decrypt':
[ 2131s] /home/abuild/rpmbuild/BUILD/john-1.9.0/src/opencl_7z_fmt_plug.c:683: undefined reference to `JOHNSWAP'
```

But that should not be part of this Pull Request, because this PR is because of general syntax issues for Big Endian.

--- a/src/opencl_pbkdf2_hmac_sha1_fmt_plug.c
+++ b/src/opencl_pbkdf2_hmac_sha1_fmt_plug.c
@@ -227,6 +227,7 @@ static void *get_binary(char *ciphertext)
 #if !ARCH_LITTLE_ENDIAN
 	char *p = strrchr(ciphertext, '$') + 1;
 	int len = strlen(p) / 2;
+	int i;
 	for (i = 0; i < len / sizeof(uint32_t); ++i) {
 		((uint32_t*)out)[i] = JOHNSWAP(((uint32_t*)out)[i]);
 	}

